src/tensors/cpu/expression_graph_packable.h

```diff
@@ -200,25 +161,81 @@ class ExpressionGraphPackable : public ExpressionGraph {
         // We need to tranpose first, because of our architecture independet format requiring a transposed matrix
         Tensor tmp;
         allocator->allocate(tmp, val->shape(), val->type());
-        Transpose10(tmp, val);
-        if (gemmElementType == Type::intgemm8) {
+        cpu::Transpose10(tmp, val);
+  
+        if(sizeOf(gemmElementType) == 1) { // is 8-bit Intgemm type
           float quantMult = 127.0f / intgemm::MaxAbsolute(val->data(), val->data() + val->shape().elements());
-          intgemm::Int8::PrepareA(tmp->data(), /*input*/
-                                paramMat->data<int8_t>(), /*output*/
-                                quantMult, /*Quant Mult*/
-                                rows(val),
-                                cols(val));
+
+          // Hardware-specific conversions which allow to implement memory-mapping and avoid conversion at runtime
+          if(isSsse3(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::SSSE3_8bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                    paramMat->data<int8_t>(), /*output*/
+                                                    quantMult, /*Quant Mult*/
+                                                    rows(val),
+                                                    cols(val));
+          } else if(isAvx2(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::AVX2_8bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                   paramMat->data<int8_t>(), /*output*/
+                                                   quantMult, /*Quant Mult*/
+                                                   rows(val),
+                                                   cols(val));
+          } else if(isAvx512(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::AVX512_8bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                     paramMat->data<int8_t>(), /*output*/
+                                                     quantMult, /*Quant Mult*/
+                                                     rows(val),
+                                                     cols(val));
+          } else {
+            ABORT_IF(gemmElementType != Type::intgemm8, "Type {} is not supported {}", gemmElementType); // shouldn't really happen, but let's make sure
```

Comment:
The surplus `{}` causes a formatting error before the abort.

Reply:
Oops. Good catch.